pkg/virt-handler/vm_test.go

```diff
@@ -804,6 +804,22 @@ var _ = Describe("VirtualMachineInstance", func() {
 			controller.Execute()
 		})
 
+		It("should move VirtualMachineInstance to Failed if configuring the networks on the virt-launcher fails", func() {
+			vmi := v1.NewMinimalVMI("testvmi")
+			vmi.ObjectMeta.ResourceVersion = "1"
+			vmi.Status.Phase = v1.Scheduled
+
+			mockWatchdog.CreateFile(vmi)
+			vmiFeeder.Add(vmi)
+
+			network.SetupPodNetworkPhase1 = func(vm *v1.VirtualMachineInstance, pid int) error { return fmt.Errorf("SetupPodNetworkPhase1 error") }
+
+			vmiInterface.EXPECT().Update(gomock.Any()).Do(func(vmi *v1.VirtualMachineInstance) {
+				Expect(vmi.Status.Phase).To(Equal(v1.Failed))
```

Comment:
I think this `Expect` should be right next to`network.SetupPodNetworkPhase1`  (line 815), with an error message 
`"..vmi status should be 'Failed.."`

Reply:
i don't understand the comment. What do you mean by right next to?

Reply:
I review it again, I guess I dont understand why line 818 is there in the first place can you please explain? :)

Reply:
After the controller finishes  [executing](https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/vm.go#L1191) on a specific vmi. If needed, it [updates the vmi via the api](https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/vm.go#L703).
This line `vmiInterface.EXPECT().Update...` verifies that the vmi that is passed to the `update` method has `v1.Failed` in the `vmi.Status.Phase` field.

That is the purpose of this test, checking that when `phase1` returns and error the vmi status is updated to be failed.